iis-workflows/iis-workflows-collapsers/src/main/resources/eu/dnetlib/iis/workflows/collapsers/origins_collapser/oozie_app/workflow.xml

```diff
@@ -18,60 +18,64 @@
 			<name>schema_input</name>
 			<description>input schema</description>
 		</property>
-        <property>
+		<property>
 			<name>output</name>
 			<description>output port</description>
 		</property>
-        <property>
+		<property>
 			<name>schema_output</name>
 			<description>output schema</description>
 		</property>
-        <property>
+		<property>
 			<name>record_collapser</name>
-            <value>eu.dnetlib.iis.workflows.collapsers.origins.OriginConfidenceCollapser</value>
+			<value>eu.dnetlib.iis.workflows.collapsers.origins.OriginConfidenceCollapser</value>
 			<description>record collapser class</description>
 		</property>
+		<property>
+			<name>mapred_child_java_opts</name>
+			<value>-Xmx2048M</value>
+			<description>java-opts, e.g. maximum heap size for oozie</description>
+		</property>
 	</parameters>
-    
-    <start to="generate-schema" />
+
+	<start to="generate-schema" />
 
 	<action name="generate-schema">
-	    <java>
-    		<job-tracker>${jobTracker}</job-tracker>
-        	<name-node>${nameNode}</name-node>
-	        <main-class>eu.dnetlib.iis.core.javamapreduce.hack.AvroSchemaGenerator</main-class>
-	        <arg>${schema_input}</arg>
-	        <arg>${schema_output}</arg>
-	        <capture-output />
-	    </java>
-	    <ok to="collapser" />
-	    <error to="fail" />
+		<java>
+			<job-tracker>${jobTracker}</job-tracker>
+			<name-node>${nameNode}</name-node>
+			<main-class>eu.dnetlib.iis.core.javamapreduce.hack.AvroSchemaGenerator</main-class>
+			<arg>${schema_input}</arg>
+			<arg>${schema_output}</arg>
+			<capture-output />
+		</java>
+		<ok to="collapser" />
+		<error to="fail" />
 	</action>
-	
+
 	<action name="collapser">
 		<map-reduce>
 			<job-tracker>${jobTracker}</job-tracker>
 			<name-node>${nameNode}</name-node>
 			<!-- The data generated by this node is deleted in this section -->
 			<prepare>
-                <delete path="${nameNode}${output}" />
+				<delete path="${nameNode}${output}" />
 				<delete path="${nameNode}${workingDir}/collapser" />
 				<mkdir path="${nameNode}${workingDir}/collapser" />
 			</prepare>
 			<configuration>
-				<!-- # This is a standard set of options that stays the same regardless 
-					of a concrete definition of map-reduce job -->
+				<!-- # This is a standard set of options that stays the same regardless of a concrete definition of map-reduce job -->
 
 				<!-- ## Various options -->
 
-                <property>
-                    <name>mapred.reduce.tasks</name>
-                    <value>36</value>
-                </property>
-                <property>
-                    <name>mapred.child.java.opts</name>
-                    <value>-Xmx2048M</value>
-                </property>
+				<property>
+					<name>mapred.reduce.tasks</name>
+					<value>36</value>
```

Comment:
Yeah, that's interesting. 

I went through svn history and found this magic number appearing in `r29647` after Dominika's `memory-related properties set` commit. It was on 2014-07-29 so it seems it was adjusted to current datanodes number. 

We could try not to hardcode it but to use placeholder and set it somewhere in top level workflow. I would also recommend creating dedicated ticket for this.